### PR TITLE
Remove unnecessary convars

### PR DESCRIPTION
**Pull request by BobTheBob9**
~~uses string names for convars (been supported since refactor but not yet used) rather than magic numbers~~ (moved to #740), and removes some convars we dont need at all

### CODE REVIEW

File: Northstar.CustomServers/mod/scripts/vscripts/sh_northstar_utils.gnut [hunk 1]
```diff
@@ -1,13 +1,7 @@
 globalize_all_functions
 
-// whether the server is a modded, northstar server
-bool function IsNorthstarServer()
-{
-	return GetConVarBool( "ns_is_modded_server" )
-}
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/sh_northstar_utils.gnut:7]: Removing this global function might be potentially bad for mod support? Idk if any mods use it, but if they do then they will break

Comment (BobTheBob9) [reply Northstar.CustomServers/mod/scripts/vscripts/sh_northstar_utils.gnut:7]: relatively sure nobody does, it's a feature that's never really been fully finished

File: Northstar.CustomServers/mod/scripts/vscripts/sh_northstar_utils.gnut [hunk 2]
```diff
@@ -1,13 +1,7 @@
 globalize_all_functions
 
-// whether the server is a modded, northstar server
-bool function IsNorthstarServer()
```
Comment (pg9182) [inline Northstar.CustomServers/mod/scripts/vscripts/sh_northstar_utils.gnut:4]: https://github.com/R2Northstar/NorthstarMods/blob/4730389c3bc0e750eab2ecdc3fd9a781f70e12a3/Northstar.Client/mod/scripts/vscripts/ui/menu_lobby.nut#L238
https://github.com/R2Northstar/NorthstarMods/blob/4730389c3bc0e750eab2ecdc3fd9a781f70e12a3/Northstar.Client/mod/scripts/vscripts/ui/openinvites.nut#L205-L206
https://github.com/R2Northstar/NorthstarMods/blob/4730389c3bc0e750eab2ecdc3fd9a781f70e12a3/Northstar.Client/mod/scripts/vscripts/ui/menu_private_match.nut#L277-L282
https://github.com/R2Northstar/NorthstarMods/blob/4730389c3bc0e750eab2ecdc3fd9a781f70e12a3/Northstar.Client/mod/scripts/vscripts/ui/menu_private_match.nut#L250-L253
https://github.com/R2Northstar/NorthstarMods/blob/4730389c3bc0e750eab2ecdc3fd9a781f70e12a3/Northstar.Client/mod/scripts/vscripts/ui/menu_mode_select.nut#L21-L22

File: Northstar.CustomServers/mod/scripts/vscripts/sh_northstar_utils.gnut (hunk 1 above)
Comment (pg9182) [reply Northstar.CustomServers/mod/scripts/vscripts/sh_northstar_utils.gnut:7]: I don't think anyone else uses it either.

**Review by pg9182 — CHANGES_REQUESTED** [review]
Still used in other places in NorthstarMods.

File: Northstar.Client/mod/scripts/vscripts/ui/menu_lobby.nut [hunk 3]
```diff
@@ -235,44 +235,21 @@ void function SetupComboButtonTest( var menu )
...
-		Hud_AddEventHandler( file.findGameButton, UIE_CLICK, OpenServerBrowser )
-	}
-	else
-	{
-		file.findGameButton = AddComboButton( comboStruct, headerIndex, buttonIndex++, "#MENU_TITLE_FIND_GAME" )
-		file.lobbyButtons.append( file.findGameButton )
-		Hud_AddEventHandler( file.findGameButton, UIE_CLICK, BigPlayButton1_Activate )
-	}
+	// server browser
+    file.findGameButton = AddComboButton( comboStruct, headerIndex, buttonIndex++, "#MENU_TITLE_SERVER_BROWSER" )
```
Comment (ASpoonPlaysGames) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_lobby.nut:239]:
```suggestion
	file.findGameButton = AddComboButton( comboStruct, headerIndex, buttonIndex++, "#MENU_TITLE_SERVER_BROWSER" )
```

File: Northstar.Client/mod/scripts/vscripts/ui/menu_mode_select.nut [hunk 4]
```diff
@@ -89,9 +83,7 @@ void function ModeButton_GetFocus( var button )
 
 	string mapName = PrivateMatch_GetSelectedMap()
 	bool mapSupportsMode = PrivateMatch_IsValidMapModeCombo( mapName, modeName )
-	if ( !mapSupportsMode && !IsNorthstarServer() )
-		Hud_SetText( nextModeDesc, Localize( "#PRIVATE_MATCH_MODE_NO_MAP_SUPPORT", Localize( GetGameModeDisplayName( modeName ) ), Localize( GetMapDisplayName( mapName ) ) ) )
-	else if ( IsFDMode( modeName ) ) // HACK!
+    if ( IsFDMode( modeName ) ) // HACK!
```
Comment (ASpoonPlaysGames) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_mode_select.nut:86]:
```suggestion
	if ( IsFDMode( modeName ) ) // HACK!
```

File: Northstar.Client/mod/scripts/vscripts/ui/openinvites.nut [hunk 5]
```diff
@@ -202,7 +202,10 @@ void function UpdateOpenInvite_Thread()
 
 void function UICodeCallback_OpenInviteUpdated()
 {
-	if ( file.openInviteVisible || IsNorthstarServer() )
+    // don't support on northstar
+    return
```
Comment (ASpoonPlaysGames) [inline Northstar.Client/mod/scripts/vscripts/ui/openinvites.nut:206]:
```suggestion
	// don't support on northstar
	return
```

**Review by ASpoonPlaysGames — CHANGES_REQUESTED** [review]
some weird spaces instead of tabs happening here

File: Northstar.Client/mod/scripts/vscripts/ui/openinvites.nut [hunk 6]
```diff
@@ -202,7 +202,10 @@ void function UpdateOpenInvite_Thread()
 
 void function UICodeCallback_OpenInviteUpdated()
```
Comment (GeckoEidechse) [inline Northstar.Client/mod/scripts/vscripts/ui/openinvites.nut:203]: This is for network invites or what? ^^

File: Northstar.Client/mod/scripts/vscripts/ui/menu_private_match.nut [hunk 7]
```diff
@@ -575,17 +565,12 @@ function UpdatePrivateMatchButtons()
...
-		if ( !IsNorthstarServer() )
-			Hud_SetLocked( file.inviteFriendsButton, true )
 	}
 	else
 	{
 		RHud_SetText( file.startMatchButton, "#START_MATCH" )
 		Hud_SetLocked( file.selectMapButton, false )
 		Hud_SetLocked( file.selectModeButton, false )
-		if ( !IsNorthstarServer() )
-			Hud_SetLocked( file.inviteFriendsButton, false )
```
Comment (GeckoEidechse) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_private_match.nut:588]: Need to double check in testing but wouldn't that mean that the button is unlocked but then inviting someone actual will just have no effect?

Comment (ASpoonPlaysGames) [reply Northstar.Client/mod/scripts/vscripts/ui/menu_private_match.nut:588]: In testing the button just doesn't exist, same as before.
In the lines of code above this, the combo button for inviting friends is added, under the same condition (`!IsNorthstarServer()`) which has also been removed in this PR

**Review by ASpoonPlaysGames — APPROVED** [review]
Code looks good, and works in testing (compiles properly, and menus all look the same as normal)

**Review by EladNLG — APPROVED** [review]
Code looks good. Did not test.

File: Northstar.CustomServers/mod.json [hunk 8]
```diff
@@ -4,15 +4,6 @@
...
 	"ConVars": [
-		{
-			"Name": "ns_lobby_type",
-			"DefaultValue": "0"
-		},
-		{
-			"Name": "ns_is_modded_server",
-			"DefaultValue": "1",
-			"Flags": 8192
-		},
```
Comment (GeckoEidechse) [inline Northstar.CustomServers/mod.json:15]: Just to be clear, as these convars are not replicated, there shouldn't be any compatibility issues with past versions when merging and releasing this as a patch release, right? ^^"

**Comment by GeckoEidechse** [comment]
Merging based reviews.

**Pull request merged**